### PR TITLE
Adding GitHub issue and PR templates

### PR DESCRIPTION
**Pull request by damieng**
Please push changes directly to the `add-issue-templates` branch and approve + merge when ready!

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+### References
+
+Please include relevant links supporting this change such as a:
+
+- support ticket
+- community post
+- StackOverflow post
+- support forum thread
+
+### Testing
```
Comment (joshcanhelp) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'm not seeing any tests in this repo, unit or otherwise.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+### Prerequisites
+
+- [ ] I have checked the [README documentation](https://github.com/auth0/angular2-jwt/blob/master/README.md).
+- [ ] I have checked the [Auth0 Community](https://community.auth0.com/) for related posts.
+- [ ] I have checked for related or duplicate [Issues](https://github.com/auth0/angular2-jwt
+/issues) and [PRs](https://github.com/auth0/angular2-jwt
+/pulls).
+- [ ] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
+- [ ] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
+- [ ] I am reporting this to the correct repository.
```
Comment (joshcanhelp) [inline .github/ISSUE_TEMPLATE.md]: May want to mention any related Angular repos

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (damieng) [reply .github/PULL_REQUEST_TEMPLATE.md]: Remove the whole section?  Might discourage people adding tests.

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (damieng) [reply .github/ISSUE_TEMPLATE.md]: Only other Angular2 one we have is archived.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (joshcanhelp) [reply .github/PULL_REQUEST_TEMPLATE.md]: I would remove anything that makes it seem like there are tests to add to. "Adding test coverage" for a library that doesn't have tests already is basically "adding a test suite," which we should probably do ourselves.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+## Documentation
+
+- PR for docs site update, if needed
+- Code-level documentation expectations
+	- 100% documentation coverage for PRs
+	- Include links to relevant Auth0 doc pages
+
+## Assets compilation
+
+Information about compiling CSS, JS, SVG, etc.
```
Comment (joshcanhelp) [inline CONTRIBUTING.md]: Is this a TODO? Seems like dummy content. I'd say remove it if we're not adding anything helpful here.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+## Environment setup
+
+Link to [README installation](README.md#installation) steps and include anything additional needed to contribute to the project.
```
Comment (joshcanhelp) [inline CONTRIBUTING.md]: Same comment here ...

**Pull request merged**